src/parser.js

```diff
@@ -938,58 +938,60 @@
      * Takes url corresponding to an SVG document, and parses it into a set of fabric objects. Note that SVG is fetched via XMLHttpRequest, so it needs to conform to SOP (Same Origin Policy)
      * @memberOf fabric
      * @param {String} url
-     * @param {Function} callback
      * @param {Function} [reviver] Method for further parsing of SVG elements, called after each fabric object created.
+     * @return {Promise} Promise which receives instance in its `then` handler
      */
-    loadSVGFromURL: function(url, callback, reviver) {
-
+    loadSVGFromURL: function(url, reviver) {
       url = url.replace(/^\n\s*/, '').trim();
-      svgCache.has(url, function (hasUrl) {
-        if (hasUrl) {
-          svgCache.get(url, function (value) {
-            var enlivedRecord = _enlivenCachedObject(value);
-            callback(enlivedRecord.objects, enlivedRecord.options);
-          });
-        }
-        else {
-          new fabric.util.request(url, {
-            method: 'get',
-            onComplete: onComplete
-          });
-        }
-      });
 
-      function onComplete(r) {
+      return new Promise(function(resolve, reject) {
+        svgCache.has(url, function (hasUrl) {
+          if (hasUrl) {
+            svgCache.get(url, function (value) {
+              var enlivedRecord = _enlivenCachedObject(value);
+              resolve(enlivedRecord.objects, enlivedRecord.options);
```

Comment:
This is not valid; a Promise can only ever be resolved with a single value. With the current approach, the second value would simply be lost. To resolve with multiple values, resolve with an object (with named keys) or an array.